conformance/utils/suite/suite.go

```diff
@@ -55,10 +70,50 @@ type ConformanceTestSuite struct {
 	FS                       embed.FS
 	UsableNetworkAddresses   []v1beta1.GatewayAddress
 	UnusableNetworkAddresses []v1beta1.GatewayAddress
+
+	// mode is the operating mode of the implementation.
+	// The default value for it is "default".
+	mode string
+
+	// implementation contains the details of the implementation, such as
+	// organization, project, etc.
+	implementation confv1b1.Implementation
+
+	// apiVersion is the version of the Gateway API installed in the cluster
+	// and is extracted by the annotation gateway.networking.k8s.io/bundle-version
+	// in the Gateway API CRDs.
+	apiVersion string
+
+	// apiChannel is the channel of the Gateway API installed in the cluster
+	// and is extracted by the annotation gateway.networking.k8s.io/channel
+	// in the Gateway API CRDs.
+	apiChannel string
+
+	// conformanceProfiles is a compiled list of profiles to check
+	// conformance against.
+	conformanceProfiles sets.Set[ConformanceProfileName]
+
+	// running indicates whether the test suite is currently running
+	running bool
```

Comment:
Can you remind me why we need this? Assuming we do, it would be useful to include the rationale in the comment for future reviewers.

Reply:
I've updated the comment:
```go
	// running indicates whether the test suite is currently running.
	// Throught this flag we prevent a Run() execution to happen in case
	// another Run() is ongoing.
	running bool
```